### PR TITLE
fix(react): Conditionally render deprecation warning

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

Only show the `Link` component's `to` prop deprecation warning if the customer does **NOT** provide an `as` prop. 

Fixes: https://github.com/aws-amplify/amplify-ui/issues/2889

#### Issue #, if available

https://github.com/aws-amplify/amplify-ui/issues/2889

#### Description of how you validated changes

Tested locally with a few examples on the docs site. 

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/Link/Link.tsx [hunk 1]
```diff
@@ -7,19 +7,19 @@ import { View } from '../View';
...
+    shouldWarn: to != null && !as,
     message:
       "The Link component's to prop will soon be deprecated. " +
       'Please see the Amplify UI documentation for using the Link component with routing libraries: ' +
       'https://ui.docs.amplify.aws/react/components/link#routing-libraries',
   });
   return (
     <View
-      as={as}
+      as={as ?? 'a'}
```
Comment (dbanksdesign) [inline packages/react/src/primitives/Link/Link.tsx:22]: Should we use a ternary here to be more explicit?

File: packages/react/src/primitives/Link/__tests__/Link.test.tsx [hunk 2]
```diff
@@ -116,10 +116,17 @@ describe('Link: ', () => {
...
     const spyWarn = jest.spyOn(console, 'warn');
     render(<Link to="/test">Test</Link>);
     expect(spyWarn).toHaveBeenCalled();
     spyWarn.mockRestore();
   });
+
+  it('should not call console.warn if "to" prop is used with "as" prop', async () => {
+    const spyWarn = jest.spyOn(console, 'warn');
+    render(<SampleRoutingApp />);
+    expect(spyWarn).not.toHaveBeenCalled();
```
Comment (dbanksdesign) [inline packages/react/src/primitives/Link/__tests__/Link.test.tsx:129]: Nice!

**Review by dbanksdesign — APPROVED** [review]
One non-blocking comment, LGTM!

File: packages/react/src/primitives/Link/Link.tsx (hunk 1 above)
Comment (reesscot) [reply packages/react/src/primitives/Link/Link.tsx:22]: +1

Comment (joebuono) [reply packages/react/src/primitives/Link/Link.tsx:22]: Like this?

`as ? as : 'a'`

**Review by reesscot — APPROVED** [review]

**Pull request merged**